### PR TITLE
Add support for --arch flag and add new OS versions to PackageDescription

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci test

File: Sources/PackageDescription/SupportedPlatforms.swift [hunk 1]
```diff
@@ -208,6 +208,19 @@ extension SupportedPlatform {
         /// - Since: First available in PackageDescription 5.1
         @available(_PackageDescription, introduced: 5.1)
         public static let v10_15: MacOSVersion = .init(string: "10.15")
+
+        /// The value that represents macOS 10.16, which is treated
+        /// as an alias for macOS 11.0.
+        ///
+        /// - Since: First available in PackageDescription 5.3
+        @available(_PackageDescription, introduced: 5.3)
```
Comment (jakepetroules) [inline Sources/PackageDescription/SupportedPlatforms.swift:216]: Can we mark this deprecated + renamed so that developers get a fixit?

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test self hosted

File: Sources/PackageDescription/SupportedPlatforms.swift (hunk 1 above)
Comment (aciidb0mb3r) [reply Sources/PackageDescription/SupportedPlatforms.swift:216]: I think that's what we want to do but as a separate commit/PR.

**Comment by aciidb0mb3r** [comment]
@swift-ci test

**Pull request merged**